### ISSUE TITLE
Android 12 Update Request - Updating Permission Check from PermissionChecker to ContextCompat

### ISSUE THREAD

**Issue by jcoona** [permissions-dispatcher/PermissionsDispatcher#756]
## Overview

- I noticed that when using `PermissionUtils.hasSelfPermissions(Context context, String... permissions)` and we include the Audio permission in this collection, it will trigger the Android 12 Privacy Indicator, and add a log to the user's privacy dashboard. See: [Article](https://9to5google.com/2021/05/18/android-12-introduces-privacy-dashboard-camera-mic-indicators-and-blocks/)
- I don't think this is really expected, as you can query a device's permissions using `ContextCompat.checkSelfPermission(Context context, String permission)` and the indicator will not be triggered. I am wondering if the following update could be made:

```
private static boolean hasSelfPermission(Context context, String permission) {
        try {
            return ContextCompat.checkSelfPermission(context, permission) == PackageManager.PERMISSION_GRANTED;
        } catch (RuntimeException t) {
            return false;
        }
    }
```

See: [Google's documentation on requesting permissions](https://developer.android.com/training/permissions/requesting)

I am requesting this feature because in my app, I am often checking to see if a permission is granted on app launch, so that if its missing, I can present the user with an instructional page on why they may want to grant the permission. If they have it granted already, we just move on. This check will falsely cause the user to think I am unexpectedly accessing the camera / microphone.

### Expected

- When querying if a device has granted permissions for an app, it should not trigger an indicator to the user that the app is accessing microphone / camera / etc. it should also not log this interaction in the user's privacy dashboard.

### Actual

- When querying if a device has granted permissions for an app, it triggers an indicator to the user that the app is accessing microphone / camera / etc. it also logs this interaction in the user's privacy dashboard. 

## Environment

- Which library version are you using? 4.6.0
- On which devices do you observe the issue? Devices running Android 12
- Note any other information that might be useful

## Reproducible steps

- `PermissionUtils.hasSelfPermissions` or `PermissionUtils.hasSelfPermission` will trigger this experience.

**Comment by hotchemi** [comment]
Thx! kinda interesting only `PermissionChecker` causes the issue but I don't have any objection to use ContextCompat.

https://github.com/permissions-dispatcher/PermissionsDispatcher/pull/757

**Comment by hotchemi** [comment]
BTW, maybe we should report to AOSP or androidx team? sounds like a bug or lack of documentation🤔 @jcoona

**Comment by crimsonwoods** [comment]
Maybe, this issue is a bug of old AndroidX Core library.

See below links:
- https://issuetracker.google.com/issues/191276590
- https://github.com/androidx/androidx/commit/4da05dd92470d2c0a9d75f5a056e42cdc6bf8ad3
- https://developer.android.com/jetpack/androidx/releases/core#1.7.0-alpha02
- https://android-review.googlesource.com/c/platform/frameworks/support/+/1787096

IMO, this issue is fixed on Android Core 1.7.0 or later, so, updating dependency on it is the first choice to fix this issue.
